src/storage/on_disk.rs

```diff
@@ -0,0 +1,242 @@
+use std::{
+    path::{Path, PathBuf},
+    sync::Mutex,
+};
+
+use ahash::AHashMap;
+use alloy_primitives::{keccak256, Address, B256, B64, U256};
+use libmdbx::{
+    Database, DatabaseKind, DatabaseOptions, Mode, NoWriteMap, ReadWriteOptions, SyncMode,
+    TableFlags, WriteFlags,
+};
+
+use super::{AccountBasic, EvmAccount, EvmCode, Storage};
+
+/// A storage that reads data from an on-disk MDBX database.
+#[derive(Debug)]
+pub struct OnDiskStorage {
+    db: Database<NoWriteMap>,
+    cache_accounts: Mutex<AHashMap<Address, Option<EvmAccount>>>,
```

Comment:
This is obviously bad for parallel as two threads reading different addresses still block on this single `Mutex`. Let's use `dashmap` for concurrent maps like in `MvMemory`.

Reply:
Also, why is the inner `Option`?

Reply:
> why is the inner Option

If there is no account associated with an address, I will store `None` in the cache so that when the same query is performed, the cached value (`None`) will be returned immediately (instead of querying the `db`).